packages/cli/src/commands/alter.test.ts

```diff
@@ -0,0 +1,225 @@
+      miscUrl: 'file:/usecannon.com/new-misc',
+    });
+
+    const mockReadDeploy = jest.fn(async (url) => {
+      switch (url) {
+        case 'file:/usecannon.com/misc':
+          return { misc: 'info' };
+        case 'file:/usecannon.com/meta':
+          return { itsMeta: 'data' };
+        case 'file:/usecannon.com':
+          return testPkgData;
+        case url:
+          return testPkgData;
+        case newUrl:
+          return newTestPkgData;
+      }
+    });
+
+    jest.spyOn(localLoader, 'read').mockImplementation(mockReadDeploy);
+
+    // Call the 'alter' function with the necessary arguments
+    await alter(packageName, chainId, preset, testPkgData.meta, command, targets, runtimeOverrides);
+
+    expect(CannonStorage.prototype.readDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(packageName, preset, chainId);
+    expect(CannonStorage.prototype.putDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(newTestPkgData);
+    expect(mockedFallBackRegistry.publish as jest.Mock<any, any>).toHaveBeenCalledWith(
+      [packageName],
+      `${chainId}-${preset}`,
+      newUrl,
+      metaUrl
+    );
+  });
+
+  test('should perform alteration for set-contract-address command', async () => {
+    // Set up test data and variables
+    const command = 'set-contract-address';
+    const targets = ['TestContract', '0x2222222222222222222222222222222222222222'];
+
+    // Call the 'alter' function with the necessary arguments
+    await alter(packageName, chainId, preset, testPkgData.meta, command, targets, runtimeOverrides);
+
+    expect(CannonStorage.prototype.readDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(packageName, preset, chainId);
+    expect(CannonStorage.prototype.putDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(testPkgData);
+    expect(testPkgData.state['provision.dummyStep'].artifacts.contracts!.TestContract.address).toEqual(targets[1]);
+    expect(mockedFallBackRegistry.publish as jest.Mock<any, any>).toHaveBeenCalledWith(
+      [packageName],
+      `${chainId}-${preset}`,
+      newUrl,
+      metaUrl
+    );
+  });
+
+  it('should perform alteration for mark-complete', async () => {
+    const command = 'mark-complete';
+    const targets = ['provision.dummyStep'];
+    const hash = '0xmark-complete-fffffffffffffffffffffffffffffffffffffffffffffffff';
+    jest.spyOn(ChainDefinition.prototype, 'getState').mockResolvedValue(hash);
+
+    // Call the 'alter' function with the necessary arguments
+    await alter(packageName, chainId, preset, testPkgData.meta, command, targets, runtimeOverrides);
+
+    expect(CannonStorage.prototype.readDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(packageName, preset, chainId);
+    expect(CannonStorage.prototype.putDeploy as jest.Mock<any, any>).toHaveBeenCalledWith(testPkgData);
+
+    // TODO: I am not sure the package status must be changed to another value
+    // expect(testPkgData.status).toEqual('complete');
```

Comment:
@dbeal-eth it looks like it would be better to modify package.stauts value, WDYT?

Reply:
I don't think it should be altered at all because this value reflects the state of whether the package is complete or not. If after marking complete, the package has all of its steps completes, the user can verify this but running the Canon build command again and it will recognize that there is no steps to run and mark the package is complete